File: src/main/kotlin/driver/BrowserFactory.java
package driver;

import com.microsoft.playwright.Browser;
import com.microsoft.playwright.BrowserType;
import com.microsoft.playwright.Playwright;
import config.Config;

public final class BrowserFactory {
    private BrowserFactory() {}

    public static Browser create(Playwright playwright) {
        BrowserType.LaunchOptions opts = new BrowserType.LaunchOptions()
                .setHeadless(Config.headless())
                .setSlowMo(Config.slowMoMs());

        String b = Config.browser().toLowerCase();

        return switch (b) {
            case "firefox" -> playwright.firefox().launch(opts);
            case "webkit" -> playwright.webkit().launch(opts);
            case "chromium" -> playwright.chromium().launch(opts);
            case "chrome" -> playwright.chromium().launch(
                    new BrowserType.LaunchOptions()
                            .setHeadless(Config.headless())
                            .setSlowMo(Config.slowMoMs())
                            .setChannel("chrome")
            );
            case "msedge", "edge" -> playwright.chromium().launch(
                    new BrowserType.LaunchOptions()
                            .setHeadless(Config.headless())
                            .setSlowMo(Config.slowMoMs())
                            .setChannel("msedge")
            );
            default -> throw new IllegalArgumentException(
                    "Unsupported browser: " + b + " (use chromium|firefox|webkit|chrome|msedge)"
            );
        };
    }
}

File: src/main/kotlin/driver/BrowserFactory.kt
package driver

import com.microsoft.playwright.Browser
import com.microsoft.playwright.BrowserType
import com.microsoft.playwright.Playwright
import config.Config

object BrowserFactory {
    fun create(playwright: Playwright): Browser {
        val commonOpts = BrowserType.LaunchOptions()
            .setHeadless(Config.headless())
            .setSlowMo(Config.slowMoMs().toDouble())

        return when (Config.browser().lowercase()) {
            "firefox" -> playwright.firefox().launch(commonOpts)
            "webkit" -> playwright.webkit().launch(commonOpts)
            "chromium" -> playwright.chromium().launch(commonOpts)
            "chrome" -> playwright.chromium().launch(
                BrowserType.LaunchOptions()
                    .setHeadless(Config.headless())
                    .setSlowMo(Config.slowMoMs().toDouble())
                    .setChannel("chrome")
            )
            "msedge", "edge" -> playwright.chromium().launch(
                BrowserType.LaunchOptions()
                    .setHeadless(Config.headless())
                    .setSlowMo(Config.slowMoMs().toDouble())
                    .setChannel("msedge")
            )
            else -> throw IllegalArgumentException("Unsupported browser: ${Config.browser()} (use chromium|firefox|webkit|chrome|msedge)")
        }
    }
}

